packages/circuit-ui/components/ColorInput/ColorInput.spec.tsx

```diff
@@ -61,4 +62,116 @@ describe('ColorInput', () => {
+        "input[type='color']",
+      ) as HTMLInputElement;
+      const newValue = '#00ff00';
+
+      fireEvent.input(colorPicker, { target: { value: newValue } });
+
+      const colorInput = container.querySelector(
+        "input[type='text']",
+      ) as HTMLInputElement;
+      expect(colorInput.value).toBe(newValue.replace('#', ''));
+    });
+
+    it('should update color input if text input changes', async () => {
+      const { container } = render(<ColorInput {...baseProps} />);
+      const colorInput = container.querySelector(
+        "input[type='text']",
+      ) as HTMLInputElement;
+      const newValue = '00ff00';
+
+      await userEvent.type(colorInput, newValue);
+
+      const colorPicker = container.querySelector(
+        "input[type='color']",
+      ) as HTMLInputElement;
+      expect(colorPicker.value).toBe(`#${newValue}`);
+    });
+  });
+
+  describe('OnChange events', () => {
+    it('should trigger onChange event when color picker changes', async () => {
+      const onChange = vi.fn();
+      const { container } = render(
+        <ColorInput {...baseProps} onChange={onChange} />,
+      );
+
+      const colorPicker = container.querySelector(
+        "input[type='color']",
+      ) as HTMLInputElement;
+
+      fireEvent.input(colorPicker, { target: { value: '#00ff00' } });
+
+      expect(onChange).toHaveBeenCalledTimes(1);
+    });
+
+    it('should trigger onChange event when color hex input changes', async () => {
+      const onChange = vi.fn();
+      const { container } = render(
+        <ColorInput {...baseProps} onChange={onChange} />,
+      );
+
+      const colorInput = container.querySelector(
+        "input[type='text']",
+      ) as HTMLInputElement;
+
+      await userEvent.type(colorInput, '00ff00');
+
+      expect(onChange).toHaveBeenCalled();
+    });
+  });
+
+  describe('Paste', () => {
+    it('should handle paste events', async () => {
+      const { container } = render(<ColorInput {...baseProps} />);
+      const colorInput = container.querySelector(
+        "input[type='text']",
+      ) as HTMLInputElement;
+
+      await userEvent.click(colorInput);
+      await userEvent.paste('#00ff00');
+
+      const colorPicker = container.querySelector(
+        "input[type='color']",
+      ) as HTMLInputElement;
+      expect(colorPicker.value).toBe('#00ff00');
+      expect(colorInput.value).toBe('00ff00');
+    });
+
+    it('should ignore invalid paste event', async () => {
+      const { container } = render(<ColorInput {...baseProps} />);
+      const colorInput = container.querySelector(
+        "input[type='text']",
+      ) as HTMLInputElement;
+
+      await userEvent.click(colorInput);
+      await userEvent.paste('oviously invalid');
+
+      const colorPicker = container.querySelector(
+        "input[type='color']",
+      ) as HTMLInputElement;
+      expect(colorPicker.value).toBe('#000000');
+      expect(colorInput.value).toBe('');
```

Comment:
Why are the values different here?

Reply:
#000000 is the default for color picker, to my understanding it can never not have a value whereas the color input can be empty - meaning no color submitted.